examples/miscellaneous/plot_outlier_detection_bench.py

```diff
@@ -0,0 +1,159 @@
+==========================================
+
+This example benchmarks outlier detection algorithms using ROC curves
+on classical anomaly detection datasets. The algorithm performance
+is assessed in an outlier detection context:
+
+1. The algorithms are trained on the whole dataset which is assumed to
+contain outliers.
+
+2. The ROC curve is computed on the same dataset using the knowledge
+of the labels.
+
+
+Interpreting the ROC plot
+-------------------------
+The algorithm performance relates to how good the true positive rate (TPR)
+is at low value of the false positive rate (FPR). The better algorithms
+have the curve on the top-left of the plot and the area under curve (AUC)
+close to 1. The diagonal dashed line represents a random classification
+of outliers and inliers.
+"""
+
+from time import time
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.neighbors import LocalOutlierFactor
+from sklearn.ensemble import IsolationForest
+from sklearn.metrics import roc_curve, auc
+from sklearn.datasets import fetch_kddcup99, fetch_covtype, fetch_openml
+from sklearn.preprocessing import LabelBinarizer
+
+print(__doc__)
+
+rng = np.random.RandomState(42)
+
+# datasets
+datasets = ["http", "smtp", "SA", "SF", "forestcover",
+            "glass", "wdbc", "cardiotocography"]
+
+# outlier detection models
+models = [
+    ("LOF", LocalOutlierFactor(n_neighbors=20, contamination="auto")),
+    ("IForest", IsolationForest(random_state=rng, contamination="auto")),
+]
+
+rows = math.ceil(len(datasets)/2)
+plt.figure(figsize=(10, rows * 3))
+for dataset_idx, dataset_name in enumerate(datasets):
+    plt.subplot(rows, 2, dataset_idx + 1)
+
+    # loading and vectorization
+    print(f"Loading {dataset_name} data")
+    if dataset_name in ["http", "smtp", "SA", "SF"]:
+        dataset = fetch_kddcup99(
+            subset=dataset_name, percent10=True, random_state=rng
+        )
+        X = dataset.data
+        y = dataset.target
+
+    print("vectorizing data")
+    if dataset_name == "SF":
+        idx = rng.choice(X.shape[0], int(X.shape[0]*0.1), replace=False)
+        X = X[idx, :]  # reduce the sample size
+        y = y[idx]
+        lb = LabelBinarizer()
+        x1 = lb.fit_transform(X[:, 1].astype(str))
+        X = np.c_[X[:, :1], x1, X[:, 2:]]
+        y = (y != b"normal.").astype(int)
+
+    if dataset_name == "SA":
+        idx = rng.choice(X.shape[0], int(X.shape[0]*0.1), replace=False)
+        X = X[idx, :]  # reduce the sample size
+        y = y[idx]
+        lb = LabelBinarizer()
+        x1 = lb.fit_transform(X[:, 1].astype(str))
+        x2 = lb.fit_transform(X[:, 2].astype(str))
+        x3 = lb.fit_transform(X[:, 3].astype(str))
+        X = np.c_[X[:, :1], x1, x2, x3, X[:, 4:]]
+        y = (y != b"normal.").astype(int)
+
+    if dataset_name == "http" or dataset_name == "smtp":
+        y = (y != b"normal.").astype(int)
+
+    if dataset_name == "forestcover":
```

Comment:
now `if dataset_name == "forestcover":` is under the whole `fetch_kddcup99` data